### PR TITLE
Added more flexible build options for external libraries

### PR DESCRIPTION
**Pull request by exception13**
Changes:
* Fixed concurrent building (added sdrbase dependency on limesuite)
* Don't get and build existed system libs if set ENABLE_EXTERNAL_LIBRARIES to AUTO
* Force use external library if ENABLE_EXTERNAL_LIBRARIES set to AUTO and  ${LIBRARY_NAME}_EXTERNAL set to ON

Example:
`cmake -DFORCE_SSE41=ON -DENABLE_EXTERNAL_LIBRARIES=AUTO -DLIBMIRISDR_EXTERNAL=ON -DLIBDSDCC_EXTERNAL=ON
`

libmirisdr and libdsdcc forced to external, other libraries will be taken from the system if possible

**WARNING: NOT TESTED ON WINDOWS AND MACOS!!!**

### CODE REVIEW

**Comment by f4exb** [comment]
Limesuite does not depend on sdrbase this is the opposite.

File: .appveyor.yml [hunk 1]
```diff
@@ -70,7 +70,7 @@ for:
             libusb-1.0-0-dev libboost-all-dev libasound2-dev libopencv-dev libopencv-imgcodecs-dev \
             libxml2-dev bison flex ffmpeg  libpostproc-dev libavcodec-dev libavformat-dev \
             libopus-dev libcodec2-dev libairspy-dev libhackrf-dev \
-            libbladerf-dev libsoapysdr-dev libiio-dev \
+            libbladerf-dev libsoapysdr-dev libiio-dev libairspyhf-dev librtlsdr-dev \
```
Comment (f4exb) [inline .appveyor.yml:73]: No: these libraries are built from source.

File: .travis.yml [hunk 2]
```diff
@@ -59,6 +59,7 @@ addons:
       - libairspy-dev
       - libhackrf-dev
       - librtlsdr-dev
+      - libairspyhf-dev
```
Comment (f4exb) [inline .travis.yml:62]: Same remark as above

File: .appveyor.yml (hunk 1 above)
Comment (exception13) [reply .appveyor.yml]: * Depends removed

**Comment by exception13** [comment]
> Limesuite does not depend on sdrbase this is the opposite.
sdrbase use limerfe. On my PC concurent build (8 threads) with external limesuite failed (at the time of build srdbase there are no headers and libraries of limesuite)

**Comment by exception13** [comment]
> Anyway it is intentional that critical dependent libraries are built from source so there will be no other options. By experience being dependent on distributed libraries brings instabilities and lack of flexibility for implementing new features.

AUTO is optional. I'm build on Debian Testing and most libraries have the latest versions and it's reduce build time.

**Comment by f4exb** [comment]
I entirely disagree on the debian/control file that should remain as-is. It is my intention not to use the packaged libraries here. Moreover Debian testing is not supported the resulting .deb is for Ubuntu 18.04 only and provided via Appveyor.

The opposite of "Limesuite depends on sdrbase" is 'sdrbase depend on LimeSuite"so we agree on this.

**Comment by exception13** [comment]
* Reverted debian/control and CI scripts changes

**Pull request merged**